### ISSUE TITLE
Add a plugin store

### ISSUE THREAD

**Issue by oddbyte** [revenge-mod/revenge-rewrite#4]
There was a plugin called Plugin Browser for bunny, but that wasn't a built-in behavior. I'd like to see a built in plugin store in the client.

**Comment by oSumAtrIX** [comment]
I'm unsure how much we can realize this because it conflicts with decentralization. At least the default source needs to be environment configurable in source.

**Comment by true1ann** [comment]
> it conflicts with decentralization.

You can just have list of repos to gather infos from, with the last in the list being the most preffered source, no?

**Comment by oSumAtrIX** [comment]
No, because then we still have a central opinionated list, regardless of ordering that conflicts with decentralization.

**Comment by Ushie** [comment]
You can be decentralised and still offer defaults, make the sources customizable and ship with our default source, users can add or delete sources as they choose

This behavior can also be found in ReVanced's upcoming update to their manager

**Comment by oSumAtrIX** [comment]
This is correct. The default can be configured via an env variable during publication. The source remains independent of any specific default.

**Comment by oddbyte** [comment]
You should also add a way to change the signing key required for plugins to be verified as "trusted"

**Comment by oSumAtrIX** [comment]
If you're using the revenge distribution of the source code, the environment will configure revenge to be trusted as part of you already trusting this distro. Changing the key doesn't make sense, however you can trust other developers keys anyway in the first place, but you can't not trust revenge if you're using revenge as the distro. If you don't want to trust revenge, dont use revenge as the distribution and compile the source with your env or someone else

**Comment by oddbyte** [comment]
You could still add the option to not trust it, and / or put your own signing key to replace Revenge's (for the like two people who are extremely privacy focused but are too lazy to recompile)

**Comment by oSumAtrIX** [comment]
This is illogical. If you run a distribution by Revenge you automatically trust it already. Not trusting the signature implies not trusting the distro as the signatory is the same. "Too lazy" and "extremely privacy focused" is mutually exclusive.
